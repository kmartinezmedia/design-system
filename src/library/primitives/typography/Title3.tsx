import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Text, TextProps } from './Text';
import {
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
} from '@designSystem/theme';

export const Title3: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 4 },
  ...textProps
}) => (
  <Text style={styles.Title3} spacing={spacing} {...textProps}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  Title3: {
    fontSize: fontSize.title3,
    lineHeight: lineHeight.title3,
    fontFamily: fontFamily.medium,
    fontWeight: fontWeight.medium,
  },
});
