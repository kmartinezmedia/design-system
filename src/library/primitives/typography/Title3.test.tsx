import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Title3 } from './Title3';

describe('Title3', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Title3>Hey, lil 🐛</Title3>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
