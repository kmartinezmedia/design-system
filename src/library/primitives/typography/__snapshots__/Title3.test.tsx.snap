// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title3 default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Medium",
        "fontSize": 20,
        "fontWeight": "500",
        "lineHeight": 28,
      },
      Object {
        "paddingBottom": 16,
      },
      Object {
        "color": "#071422",
        "lineHeight": 28,
        "overflow": "visible",
        "textAlign": undefined,
      },
      undefined,
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
